### PR TITLE
Switch from Chrome to Edge driver for Selenium Tests

### LINKED ISSUE #15527: Enable Caseflow test suite to run on Microsoft Edge
## Description
The VA will soon be setting Microsoft Edge as the default browser on VA computers. In preparation to support Caseflow on Microsoft Edge, we need to enable our automated tests to run on the Edge browser instead of Chrome to see if there are any gaps.

Here are some initial instructions: https://docs.microsoft.com/en-us/microsoft-edge/webdriver-chromium?tabs=c-sharp

## Acceptance criteria
- [ ] Updated local test suite to run on Microsoft Edge
- [ ] Update Circle CI tests to run on Microsoft Edge (if possible)
- [ ] If there are any test failures, [record them in this Google doc](https://docs.google.com/document/d/1jjVdbRnBsXwZXrUbOFxZrOIgt0UBXvhcAcVmOzqCJIY/edit#heading=h.4v5339q3b9lz) and hold off on merging the PR

## Background/context/resources
- [Caseflow - Microsoft Edge Testing Plan](https://docs.google.com/document/d/1jjVdbRnBsXwZXrUbOFxZrOIgt0UBXvhcAcVmOzqCJIY/edit)

## Technical Notes

- It appears that this would just require updating the [Capybara Selenium Driver](https://github.com/department-of-veterans-affairs/caseflow/blob/5d0bf14c9747fc9440f70afcdcebe8a38c3f9a8e/spec/support/capybara.rb#L22) to use Edge instead of using Chrome (See [this thread](https://stackoverflow.com/questions/60940488/how-do-i-set-options-for-the-selenium-ms-edge-driver-in-ruby#answer-60958149))

### PR DESCRIPTION
**Pull request by ferristseng**
Resolves #15527

### Description

- Dependency upgrade
  - **Upgrade to Selenium 4** ([this is the required to use webdrivers with Edge](https://github.com/titusfortner/webdrivers#microsoft-edge-chromium))
  - Upgrade `webdrivers` to `4.4.*`
  - Upgrade `capybara` and `capybara=snapshot`
  - Upgrade `bootsnap` to `1.7` to resolve a startup issue with the new CircleCI image
  - **Upgrade bundler to 2.2.7** because bundler 1 is not available on the new CircleCI image
- Update CircleCI base image to use the CircleCI's next-gen images
  - `pdftk` is not available on Ubuntu 18.04, so the image has been updated to build and install it
  - **Install Microsoft Edge** on CircleCI image
- Update CircleCI config to use [browser-tools orb](https://circleci.com/developer/orbs/orb/circleci/browser-tools) (only necessary to install Chrome for the Chrome feature tests)
- Add a new job to run Chrome feature tests
- Update `rspec` job to run feature tests with Edge
- Update feature tests to run with Edge by default
- Resolve test breakages

**TODO:**

- [x] Figure out how to deal with Selenium path (set `WD_EDGE_CHROME_PATH` according to code)
- [x] Find a CircleCI image or create one that has Edge installed (see `ci-bin/circle_docker_container`)
  - [x] Update Dockerfile to install Edge
- [x] Setup another Knapsack Pro API Token for the feature tests suite (already created it)
- [x] Add an environment variable that lets you specify the feature test suite webdriver
- [x] Switch build flag to target only master branch
- [x] Publish CircleCI docker image under a new tag name (the new image breaks compatibility and shouldn't be pulled in to branches that require the old image)

### Developer Changes

These changes will be necessary for developers after this gets merged:

- Upgrade to Bundler 2.2.7
- bundle install
- Setup Edge locally

### CODE REVIEW

**Comment by alisan16** [comment]
Is there a way to keep both Chrome driver and Edge driver tests for the Caseflow master branch? Having just one for each develop branch would be good for time, but there may be a temporary period where we are supporting both Chrome and Edge in production

**Comment by ferristseng** [comment]
> Is there a way to keep both Chrome driver and Edge driver tests for the Caseflow master branch? Having just one for each develop branch would be good for time, but there may be a temporary period where we are supporting both Chrome and Edge in production

Yes, I think there could be. One possible way to achieve this (still need to research if this is possible):

  1. Use an environment variable to control which web driver to use
  2. Have an additional step in CircleCI that runs for the master branch that only runs feature tests using Chrome

**Comment by ferristseng** [comment]
Note to self: The upgraded version of Selenium requires Chrome 75+. The image is built with 72

```
circleci@778ba9a152ff:~/project$ bundle exec rake webdrivers:chromedriver:version
2021-01-15 23:08:40 INFO Webdrivers chromedriver 72.0.3626.69
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -373,5 +484,12 @@ workflows:
       - rspec:
           requires:
             - build_workspace
+      - rspec_feature_tests_chrome:
+          requires:
+            - build_workspace
+          filters:
+            branches:
+              ignore:
```
Comment (ferristseng) [inline .circleci/config.yml:492]: Change this to say `only` so the Chrome feature tests only run on master

**Comment by ferristseng** [comment]
Related Docs: https://github.com/department-of-veterans-affairs/caseflow/pull/15852
